src/libfetchers/fetchers.cc

```diff
@@ -257,6 +257,13 @@ std::optional<time_t> Input::getLastModified() const
     return {};
 }
 
+std::optional<std::string> Input::getModules() const
+{
+    if (auto s = maybeGetStrAttr(attrs, "modules"))
```

Comment:
I'm surprised that this works... Isn't modules an attrset? Doesn't `maybeGetStrAttr` throw if the attribute isn't a string?

Reply:
It's already stored as JSON. `fetchTree` reconstructs it into an attrset here: https://github.com/NixOS/nix/pull/5497/files#diff-e473b01a82cc31e51baa89b26b4fe5083e56df27cd834b9722789841ef9d5620R38-R50

not the prettiest thing in the world but it works 🤷‍♀️

Reply:
:scream: 

Does the process producing the JSON sort keys?

Reply:
oops, you're right. we only sort in `fetchTree` 😅

Reply:
Hmmm... looks like nlohmann takes care of it, unless I'm missing something: https://json.nlohmann.me/features/object_order/